main/main.go

```diff
@@ -232,8 +245,101 @@ func main() {
 		manualWarpMessages[sourceBlockchainID] = append(manualWarpMessages[sourceBlockchainID], &warpLogInfo)
 	}
 
-	// Create listeners for each of the subnets configured as a source
 	errGroup, ctx := errgroup.WithContext(context.Background())
+
+	if len(cfg.DeciderPluginPath) > 0 {
+		var cancel context.CancelFunc
+		args := make([]string, 0)
+		cmdCtx, cancel := context.WithCancel(ctx)
+		cmd := exec.CommandContext(cmdCtx, cfg.DeciderPluginPath, args...)
+		cmdStdOutReader, err := cmd.StdoutPipe()
+		if err != nil {
+			logger.Error(
+				"Failed to pipe stdout from decider plugin process",
+				zap.Error(err),
+			)
+			panic(err)
+		}
+		cmdStdErrReader, err := cmd.StderrPipe()
+		if err != nil {
+			logger.Error(
+				"Failed to pipe stdout from decider plugin process",
+				zap.Error(err),
+			)
+			panic(err)
+		}
+		// Start goroutines to read and output the command's stdout and stderr
+		stopScanning := make(chan bool)
+		errGroup.Go(func() error {
+			scanner := bufio.NewScanner(cmdStdOutReader)
+			for scanner.Scan() {
+				logger.Info(scanner.Text())
+			}
+			stopScanning <- true
+			return nil
+		})
+		errGroup.Go(func() error {
+			scanner := bufio.NewScanner(cmdStdErrReader)
+			for scanner.Scan() {
+				logger.Error(scanner.Text())
+			}
+			stopScanning <- true
+			return nil
+		})
+		errGroup.Go(func() error {
+			err = cmd.Start()
+			if err != nil {
+				logger.Error("Failed to start decider process")
+				return err
+			}
+			grpcClient, err = grpc.NewClient(
+				"127.0.0.1:50051",
+				grpc.WithTransportCredentials(insecure.NewCredentials()),
+			)
+			if err != nil {
+				logger.Fatal(
+					"Could not connect to decider process",
+					zap.Error(err),
+				)
+				panic(err)
+			}
+			logger.Info("Decider running")
+			<-ctx.Done()
+			logger.Info("global context cancelled, cancelling decider")
+			<-stopScanning
+			cancel()
+			<-cmdCtx.Done()
+			logger.Info("waiting for decider to finish")
```

Comment:
this is a mess of experimentation, and probably incorrect, but it's because I've yet to actually exercise a graceful shutdown of the process, because I think awm-relayer itself isn't being shut down gracefully. this mess should clean itself up easily once we actually start running through this code.